### PR TITLE
Add utility method to format phone numbers

### PR DESCRIPTION
**Pull request by mkrausse-ggtx**
This pull request adds a new utility method, `format_phone_number`, which formats a phone number in E.164 format. The method takes a phone number as input and an optional country code, and returns the formatted phone number. It removes non-numeric characters and leading zeros, adds the country code prefix if necessary, and formats the number in E.164 format. The pull request also includes unit tests for the `format_phone_number` method to ensure its correctness and handle different scenarios.

### CODE REVIEW

**Comment by austinweisgrau** [comment]
IMO a better default behavior for a format phone number method would be to return None for an invalid phone number rather than raise an error.

**Comment by mkrausse-ggtx** [comment]
That makes sense to me. I can update that.

**Comment by mkrausse-ggtx** [comment]
I'm happy to do either one. Do folks think my approach to not adding another dependency makes sense? I'm also willing to write a function that uses the phonenumbers package.

**Comment by austinweisgrau** [comment]
Yeah I think your approach is good! The dependency seems unnecessary.

A few other validations worth incorporating potentially - valid US phones can't have an area code that starts with 0 or 1, and the first digit after the area code also can't be 0 or 1. I think 999 is also an invalid area code.

FWIW the phonenumbers package doesn't invalidate those patterns so there's no advantage there either.

File: parsons/utilities/format_phone_number.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import re
+
+
+def format_phone_number(phone_number, country_code="1"):
+    """
+    Formats a phone number in E.164 format.
```
Comment (Jason94) [inline parsons/utilities/format_phone_number.py:6]: Could you add some documentation showing what it'll do? That would be helpful to someone who doesn't know what the e.164 format is, and might have no idea what this does.

Even one line like:

`Example: Converts "(123) 123-1212" to "+01231231212"`

would be very helpful!

Comment (matthewkrausse) [reply parsons/utilities/format_phone_number.py]: Sounds good. Went ahead and made that change.

**Pull request merged**